src/lessons/Data-Structures/04.md

```diff
@@ -0,0 +1,51 @@
+
+So far we've exclusively been discussing adorable images for the sake of our general
+merriment, but content addressing can be used on all different types of files and data,
+from JSON objects to term papers to videos. For cryptographic hashing to work, we need
+to know what data format we're working with and use an appropriate tool.
+
+## Decoding data structures
+
+A `CID` is a particular form of content addressing. It's a specification developed for
+IPFS, a specific decentralized web protocol that we'll discuss in later tutorials,
+but it has very broad implications.
+
+A `CID` is a single identifier that contains a cryptographic hash **and**
+information about how to interpret that data called a "codec." (Codecs encode,
+decode, compress, and decompress data in certain formats. You use them every time
+you stream a YouTube video of an adorable beagle.)
```

Comment:
Video codecs, like those on YouTube, are a different beast, so I'd leave this out.

In fact, we already have be aware of video codecs when encoding files in IPFS, just at a different layer (chunking).